lib/PuppeteerSharp/Launcher.cs

```diff
@@ -168,17 +172,16 @@ public async Task<Browser> ConnectAsync(ConnectOptions options)
 
                 var response = await _connection.SendAsync<GetBrowserContextsResponse>("Target.getBrowserContexts");
 
-                return await Browser.CreateAsync(_connection, response.BrowserContextIds, options.IgnoreHTTPSErrors, true, null, () =>
+                return await Browser.CreateAsync(_connection, response.BrowserContextIds, options.IgnoreHTTPSErrors, true, null, async () =>
```

Comment:
I am not sure I understand this comment. The `Browser` class takes a `Func<Task>` argument, which is what is created here. The `async () => …` syntax is needed to allow use of `await` and enable logging of close exceptions.

Reply:
Here I'm speaking more from previous experiences and pain :p 
`Browser.CloseAsync` is quite tricky. It behaves nicely when you call it explicitly. But it gave us some headaches when it's being called from the [Dispose](https://github.com/kblok/puppeteer-sharp/blob/master/lib/PuppeteerSharp/Browser.cs#L390), because of the `GetAwaiter().GetResult()` so we need there to get a task. If we can we will await it, if we can't we will fire-and-forget.

You can leave it and try but your test on the CI will get stuck.

Reply:
I'd go for the fire-and-forget then in the Dispose().